File: python/ABSTRACT_FACTORY_DECORATOR/ABSTRACT_FACTORY_DECORATOR/decorator/ConcreteDecoratorReservacionVIP.py
from ABSTRACT_FACTORY_DECORATOR.reservacion.ReservacionVIP import ReservacionVIP
from .ReservacionDecoratorFactory import ReservacionDecoratorFactory

class ConcreteDecoratorReservacionVIP(ReservacionDecoratorFactory):
    
    reservacionExtendida : ReservacionVIP
    
    def __init__(self, reservacionSencilla) -> None:
        super().__init__(reservacionSencilla)
        
    def aumentarReservacion(self, datos):
        self.volverReservacionVIP(datos)
    
    def volverReservacionVIP(self, datos): 
        self.reservacionExtendida = ReservacionVIP([self.reservacionSencilla.datosCliente,
                                                      self.reservacionSencilla.datosCompra,
                                                      self.reservacionSencilla.datosFuncion,
                                                      datos])

File: php/ABSTRACT_FACTORY_DECORATOR/decorator/ConcreteDecoratorReservacionVIP.php
<?php
require_once 'ReservacionDecoratorFactory.php';
require_once './reservacion/ReservacionVIP.php';

class ConcreteDecoratorReservacionVIP extends ReservacionDecoratorFactory{
    
    public $reservacionExtendida;
    
    public function __construct($reservacionSencilla) {
        parent::__construct($reservacionSencilla);
    }
    
    public function volverReservacionVIP($datos) {
        $this->reservacionExtendida = new ReservacionVIP($datos, $this->reservacionSencilla->datosCliente,
                $this->reservacionSencilla->datosCompra,$this->reservacionSencilla->datosFuncion);
        
    }
    
    public function aumentarReservacion($datos) {
        $this->volverReservacionVIP($datos);
    }
}
